ios/browser/api/web_image/web_image.mm

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/web_image/web_image.h"
+#include "brave/ios/browser/api/web_image/image_downloader.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "net/base/mac/url_conversions.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "ui/gfx/image/image.h"
+#include "ui/gfx/image/image_skia.h"
+#include "ui/gfx/image/image_skia_rep_ios.h"
+#include "url/gurl.h"
+
+#include <memory>
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+@interface WebImageDownloader () {
+  std::unique_ptr<brave::ImageDownloader> image_fetcher_;
+}
+@end
+
+@implementation WebImageDownloader
+- (instancetype)initWithBrowserState:(ChromeBrowserState*)browserState {
+  if ((self = [super init])) {
+    image_fetcher_ = std::make_unique<brave::ImageDownloader>(
+        browserState->GetSharedURLLoaderFactory());
+  }
+  return self;
+}
+
+- (void)dealloc {
+  image_fetcher_.reset();
```

Comment:
Is this necessary? Shouldn't it reset automatically?

Reply:
Not necessary. I usually add it anyway. But fixed.